lib/util/notification_type.dart

```diff
@@ -0,0 +1,367 @@
+        return 'new_postcard_trip';
+      case NotificationType.postcardShareExpired:
+        return 'postcard_share_expired';
+      case NotificationType.customerSupportNewMessage:
+        return 'customer_support_new_message';
+      case NotificationType.customerSupportCloseIssue:
+        return 'customer_support_close_issue';
+      case NotificationType.customerSupportNewAnnouncement:
+        return 'customer_support_new_announcement';
+      case NotificationType.artworkCreated:
+        return 'artwork_created';
+      case NotificationType.artworkReceived:
+        return 'artwork_received';
+      case NotificationType.newMessage:
+        return 'new_message';
+      case NotificationType.jgCrystallineWorkHasArrived:
+        return 'jg_artwork_solar_day_arrived';
+      case NotificationType.jgCrystallineWorkGenerated:
+        return 'jg_artwork_generated';
+      case NotificationType.unknown:
+        return 'unknown';
+    }
+  }
+
+  // fromString method
+  static NotificationType fromString(String value) {
+    switch (value) {
+      case 'gallery_new_nft':
+        return NotificationType.galleryNewNft;
+      case 'new_postcard_trip':
+        return NotificationType.newPostcardTrip;
+      case 'postcard_share_expired':
+        return NotificationType.postcardShareExpired;
+      case 'customer_support_new_message':
+        return NotificationType.customerSupportNewMessage;
+      case 'customer_support_close_issue':
+        return NotificationType.customerSupportCloseIssue;
+      case 'customer_support_new_announcement':
+        return NotificationType.customerSupportNewAnnouncement;
+      case 'artwork_created':
+        return NotificationType.artworkCreated;
+      case 'artwork_received':
+        return NotificationType.artworkReceived;
+      case 'new_message':
+        return NotificationType.newMessage;
+      case 'jg_artwork_solar_day_arrived':
+        return NotificationType.jgCrystallineWorkHasArrived;
+      case 'jg_artwork_generated':
+        return NotificationType.jgCrystallineWorkGenerated;
+      default:
+        return NotificationType.unknown;
+    }
+  }
+}
+
+class NotificationHandler {
+  // singleton
+  static final NotificationHandler instance = NotificationHandler._();
+
+  NotificationHandler._();
+
+  final ConfigurationService _configurationService =
+      injector<ConfigurationService>();
+  final RemoteConfigService _remoteConfig = injector<RemoteConfigService>();
+  final ClientTokenService _clientTokenService = injector<ClientTokenService>();
+  final NavigationService _navigationService = injector<NavigationService>();
+
+  Future<void> handleNotificationClicked(BuildContext context,
+      OSNotification notification, PageController? pageController) async {
+    if (notification.additionalData == null) {
+      // Skip handling the notification without data
+      return;
+    }
+
+    final remoteConfig = injector<RemoteConfigService>();
```

Comment:
you have 
`final RemoteConfigService _remoteConfig = injector<RemoteConfigService>();`